application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+package org.togetherjava.tjbot.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * used to log activities in the mod audit log channel
+ */
+public enum ModAuditLogWriter {
+    ;
+    private static final Color EMBED_COLOR = Color.decode("#3788AC");
+
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static final Predicate<String> isAuditLogChannelName =
+            Pattern.compile(Config.getInstance().getModAuditLogChannelPattern()).asMatchPredicate();
+    private static final Predicate<TextChannel> isAuditLogChannel =
+            channel -> isAuditLogChannelName.test(channel.getName());
+
+    /**
+     * logs an entry in the mod audit log channel.
+     *
+     * @param title the title of the log embed
+     *
+     * @param description the description of the log embed
+     *
+     * @param author the user who triggered the action
+     *
+     * @param timestamp the timestamp of the action. usually use
+     *        {@link GenericInteractionCreateEvent#getTimeCreated()}
+     *
+     * @param guild the guild. usually use {@link GenericInteractionCreateEvent#getGuild()}
+     *
+     * @param attachments the attachments that'll be added to the message
+     */
+    public static void writeModAuditLog(@NotNull String title, @NotNull String description,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull Guild guild,
+            @NotNull List<@NotNull Attachment> attachments) {
```

Comment:
`List<@NotNull Attachment>`, is that a thing? never seen it lol

Reply:
On that note, now that I see this static method in the context of the class name, `ModAuditLogWriter.writeModAuditLog(...)`, the method name starts to feel really bloated and obsolete. Lets maybe go back to just `write(...)` as the method name.

Reply:
not sure, I think I did that to prevent elements of the list to be null. when I think about it, can a List contain null objects by default?

Reply:
anyways, e52d974401ccc7be9b1c7de0ede6a270b7ab2d54 no longer uses a `List`.